cc/mcts_player.cc

```diff
@@ -197,10 +197,11 @@ Coord MctsPlayer::PickMove() {
   return c;
 }
 
-absl::Span<MctsNode* const> MctsPlayer::TreeSearch(int batch_size) {
+absl::Span<MctsNode* const> MctsPlayer::TreeSearch() {
+  int batch_size = options_.batch_size;
   int max_iterations = batch_size * 2;
 
-  leaves_.clear();
+  leaves_.resize(0);
```

Comment:
consider clear() instead of resize() which may avoid repeated mallocs

Reply:
std::vector<>::clear() does not specify whether memory is deallocated or not. Most STL implementations don't, but MSVC's does. You should not use clear() if care about memory allocation.

std::vector<int> vec(n);
vec.resize(0); // doesn't deallocate.

std::vector<int>().swap(vec); // deallocates.
// .. or in C++11, easier to read but two lines:
vec.resize(0);
vec.shrink_to_fit();